testkit/src/main/scala/akka/kafka/testkit/scaladsl/KafkaSpec.scala

```diff
@@ -55,6 +56,10 @@ abstract class KafkaSpec(val kafkaPort: Int, val zooKeeperPort: Int, actorSystem
     extends TestKit(actorSystem)
     with KafkaTestKit {
 
+  // children should setup() and use admin or oldAdmin instead of constructors
+  private def adminClient(): AdminClient = super.adminClient()
```

Comment:
@2m Is there a better way of removing visibility to children in Scala?

Reply:
Renaming the factory methods in `internal.KafkaTestKit` to `createXyz` is the best way, I guess as it indicates the caller has some responsibility.